src/crawlee/http_crawler/_http_crawler.py

```diff
@@ -127,7 +127,11 @@ async def _handle_blocked_request(self, context: HttpCrawlingContext) -> AsyncGe
         if self._retry_on_blocked:
             status_code = context.http_response.status_code
 
-            if context.session and context.session.is_blocked_status_code(status_code=status_code):
+            if (
+                context.session
+                and status_code not in self._http_client._ignore_http_error_status_codes  # noqa: SLF001
+                and context.session.is_blocked_status_code(status_code=status_code)
+            ):
```

Comment:
Wondering whether this part should not look more like this:

```python
if context.session and context.session.is_blocked_status_code(
    status_code=status_code,
    additional_blocked_status_codes=self._http_client.additional_blocked_status_codes,
    ignore_http_error_status_codes=self._http_client.ignore_http_error_status_codes
):
```

which would require further changes as well, but since you're touching it here, I'm mentioning it ... Maybe do it in other PR. If I am not overlooking anything.